actions/send_file_to_webhook_MOD.js

```diff
@@ -0,0 +1,122 @@
+
+  subtitle (data) {
+    return 'Send a file to a webhook.'
+  },
+
+  // ---------------------------------------------------------------------
+  // Action Storage Function
+  //
+  // Stores the relevant variable info for the editor.
+  // ---------------------------------------------------------------------
+
+  variableStorage (data, varType) {
+    const type = parseInt(data.storage)
+    if (type !== varType) return
+    const dataType = 'Number'
+    return ([data.varName, dataType])
+  },
+
+  // ---------------------------------------------------------------------
+  // Action Fields
+  //
+  // These are the fields for the action. These fields are customized
+  // by creating elements with corresponding IDs in the HTML. These
+  // are also the names of the fields stored in the action's JSON data.
+  // ---------------------------------------------------------------------
+
+  fields: ['url', 'filename'],
+
+  // ---------------------------------------------------------------------
+  // Command HTML
+  //
+  // This function returns a string containing the HTML used for
+  // editting actions.
+  //
+  // The "isEvent" parameter will be true if this action is being used
+  // for an event. Due to their nature, events lack certain information,
+  // so edit the HTML to reflect this.
+  //
+  // The "data" parameter stores constants for select elements to use.
+  // Each is an array: index 0 for commands, index 1 for events.
+  // The names are: sendTargets, members, roles, channels,
+  //                messages, servers, variables
+  // ---------------------------------------------------------------------
+
+  html (isEvent, data) {
+    return `
+<div style="width: 90%;">
+  Webhook URL:<br>
+  <input id="url" class="round" type="text">
+</div><br>
+<div style="width: 65%;">
+  File Name:<br>
+  <input id="filename" class="round" placeholder="../yourfilename.png" type="text">
+</div><br>
+</div>`
+  },
+
+  // ---------------------------------------------------------------------
+  // Action Editor Init Code
+  //
+  // When the HTML is first applied to the action editor, this code
+  // is also run. This helps add modifications or setup reactionary
+  // functions for the DOM elements.
+  // ---------------------------------------------------------------------
+
+  init () {},
+
+  // ---------------------------------------------------------------------
+  // Action Bot Function
+  //
+  // This is the function for the action within the Bot's Action class.
+  // Keep in mind event calls won't have access to the "msg" parameter,
+  // so be sure to provide checks for variable existance.
+  // ---------------------------------------------------------------------
+
+  action (cache) {
+    const data = cache.actions[cache.index]
+    const url = this.evalMessage(data.url, cache)
+    const filename = this.evalMessage(data.filename, cache)
+    const Mods = this.getMods()
+
+    const { Webhook } = Mods.require('discord-webhook-node')
```

Comment:
no

Reply:
try doing this shit without using outside packages first. pretty sure discord.js already fully supports webhooks

Reply:
ok danno